### PR TITLE
DOP-1523: Permit version admonitions to have no argument

### PR DESCRIPTION
**Pull request by i80and**
This change scares me -- admonitions are a very gnarly and fragile part of docutils. We have a lot of test cases, but likely not every single situation a writer has ever come up with.

### CODE REVIEW

File: snooty/rstparser.py [hunk 1]
```diff
@@ -769,7 +769,7 @@ class BaseVersionDirective(docutils.parsers.rst.Directive):
     subsequent indented directives are included as children of the node.
```
Comment (sophstad) [inline snooty/rstparser.py:769]: Docstring could probably stand to be changed:

> These directives include one required argument

No longer holds!

**Review by sophstad — APPROVED** [review]
Agree that this change is a bit 😳, but LGTM!

**Review by schmalliso — COMMENTED** [review]
+1 to Sophie's comment about the docstring.

Does this change make including the version argument optional for versionadded and versionchanged *as well as* deprecated? Because ideally, it's only be deprecated…

**Comment by i80and** [comment]
Made only `deprecated`'s argument optional, and added some more tests just to make myself feel better!

**Review by schmalliso — APPROVED** [review]
MAGNIFICO!

**Pull request merged**